plugins/woocommerce-admin/client/products/sections/images-section.tsx

```diff
@@ -0,0 +1,191 @@
+import { Icon, trash } from '@wordpress/icons';
+import { MediaItem } from '@wordpress/media-utils';
+
+/**
+ * Internal dependencies
+ */
+import { ProductSectionLayout } from '../layout/product-section-layout';
+import DragAndDrop from '../images/drag-and-drop.svg';
+import './images-section.scss';
+
+type Image = MediaItem & {
+	src: string;
+};
+
+export const ImagesSection: React.FC = () => {
+	const { getInputProps, setValue } = useFormContext< Product >();
+	const images = ( getInputProps( 'images' ).value as Image[] ) || [];
+	const [ showRemoveZone, setShowRemoveZone ] = useState( false );
+	const [ imageIdToRemove, setImageIdToRemove ] = useState< number >( 0 );
+
+	const getUniqueImages = ( files: Image[] ) => {
+		if ( ! files ) {
+			return [];
+		}
+		files.forEach( ( image: Image ) => {
+			if (
+				image.id &&
+				images.find( ( file ) => file.id === image.id ) === undefined
+			) {
+				images.push( image );
+			}
+		} );
+		setValue( 'images', images );
+		return images;
+	};
+
+	const toggleRemoveZone = () => {
+		setShowRemoveZone( ! showRemoveZone );
+	};
+
+	const setImageToRemove = ( image: string ) => {
+		const tempNode = document.createElement( 'div' );
+		tempNode.innerHTML = image;
+		const imageId = tempNode.getElementsByTagName( 'img' )[ 0 ].id;
+		setImageIdToRemove( parseInt( imageId, 10 ) );
+	};
+
+	const orderImages = ( newOrder: JSX.Element[] ) => {
+		const orderedImages = newOrder.map( ( image ) => {
+			return images.find(
+				( file ) =>
+					file.id === parseInt( image?.props?.children?.key, 10 )
+			);
+		} );
+		setValue( 'images', orderedImages );
+	};
+
+	return (
+		<ProductSectionLayout
+			title={ __( 'Images', 'woocommerce' ) }
+			description={
+				<>
+					<span>
+						{ __(
+							'For best results, use JPEG files that are 1000 by 1000 pixels or larger.',
+							'woocommerce'
+						) }
+					</span>
+					<Link
+						className="woocommerce-form-section__header-link"
+						href="https://woocommerce.com/posts/fast-high-quality-product-photos/"
+						target="_blank"
+						type="external"
+						onClick={ () => {
+							recordEvent( 'prepare_images_help' );
+						} }
+					>
+						{ __( 'How to prepare images?', 'woocommerce' ) }
+					</Link>
+				</>
+			}
+		>
+			<Card
+				className={ classnames( 'woocommerce-product-form__images', {
+					'has-images': images.length > 0,
+				} ) }
+			>
+				<CardBody>
+					<ImageGallery
+						onDragStart={ toggleRemoveZone }
+						onDragEnd={ () => {
+							if ( imageIdToRemove ) {
+								setValue(
+									'images',
+									images.filter(
+										( img ) => img.id !== imageIdToRemove
+									)
+								);
+								setImageIdToRemove( 0 );
+							}
+							toggleRemoveZone();
+						} }
+						onOrderChange={ orderImages }
+					>
+						{ images.map( ( image ) => (
+							<ImageGalleryItem
+								key={ image.id }
+								alt={ image.alt }
+								src={ image.url || image.src }
+								id={ `${ image.id }` }
```

Comment:
Note that we could forego the ID prop here and just opt to use the `key` to keep props down if we implement my above suggestion.

Reply:
Right now we're not presenting the key when we render [an `ImageGalleryItem` item](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/components/src/image-gallery/image-gallery-item.tsx#L24-L27). The rendered id is what we use to pair our list when the items are reordered or to delete an image.